src/GraphQL/GraphQLBuilderExtensions.cs

```diff
@@ -1029,13 +1029,52 @@ public static IGraphQLBuilder AddValidationRule<TValidationRule>(this IGraphQLBu
         }
         #endregion
 
-        #region - AddMetrics -
+        #region - AddApolloTracing / AddMetrics -
+        /// <summary>
+        /// Registers <see cref="InstrumentFieldsMiddleware"/> within the dependency injection framework and
+        /// configures it to be installed within the schema, and configures responses to include Apollo
+        /// Tracing data when enabled via <see cref="ExecutionOptions.EnableMetrics"/>.
+        /// When <paramref name="enable"/> is <see langword="true"/>, configures execution to set
+        /// <see cref="ExecutionOptions.EnableMetrics"/> to <see langword="true"/>; otherwise leaves it unchanged.
+        /// </summary>
+        public static IGraphQLBuilder AddApolloTracing(this IGraphQLBuilder builder, bool enable = true)
+            => AddApolloTracing(builder, _ => enable);
+
+        /// <summary>
+        /// Registers <see cref="InstrumentFieldsMiddleware"/> within the dependency injection framework and
+        /// configures it to be installed within the schema, and configures responses to include Apollo
+        /// Tracing data when enabled via <see cref="ExecutionOptions.EnableMetrics"/>.
+        /// Configures execution to run <paramref name="enablePredicate"/> and when <see langword="true"/>, sets
+        /// <see cref="ExecutionOptions.EnableMetrics"/> to <see langword="true"/>; otherwise leaves it unchanged.
```

Comment:
> otherwise leaves it unchanged

🤔 Are you sure? Why not to set it to `false` if one wants to disable metrics to some subset of requests with default metrics enabled?

Reply:
Yes I am sure.

Reply:
What is the answer for second question?

Reply:
Please read my thought process below.

As for your second question, it does not make sense.  The default value is already false.  If it was set to true manually by user code it should not be changed by the code here; they can modify their other code so that it is set to true conditionally.  If you wish to post a code snippet of a realistic scenario you anticipate happening, we can review.  I have demonstrated a number of scenarios below and I believe the behavior in this PR is correct.

Reply:
I see. Just reading emails in order.